src/IO/Importers/ReadSpecThirdOrderPiecewisePolynomial.hpp

```diff
@@ -22,15 +22,19 @@
 #include "IO/H5/Dat.hpp"
 #include "IO/H5/File.hpp"
 #include "IO/Importers/Tags.hpp"
+#include "IO/Observer/ArrayComponentId.hpp"
+#include "Parallel/ArrayIndex.hpp"
 #include "Parallel/GlobalCache.hpp"
+#include "Parallel/Invoke.hpp"
+#include "Parallel/NodeLock.hpp"
 #include "ParallelAlgorithms/Initialization/MergeIntoDataBox.hpp"
 #include "Utilities/Gsl.hpp"
 #include "Utilities/Requires.hpp"
 #include "Utilities/TMPL.hpp"
 #include "Utilities/TaggedTuple.hpp"
 
 namespace importers {
-namespace Actions {
+namespace ThreadedActions {
```

Comment:
What I want to do is run the action once per node, then broadcast the data read to all processes. Previously, this action had no way to do this, but instead ran the action once / element. This led to hdf5 read errors in my BBH runs.

Since I didn't understand spectre's parallel infrastructure that well yet, I looked in the code for an example that read in data once/node, and the numeric initial data reader seemed the closest task for my purpose: read in data once/node one time at the beginning of the simulation. So I reused as much of the numeric initial data reader code as possible.

If there's a quick way to change this PR to get the same result with a better design, I'm all for it! But if it would take a lot of work to redesign, I'd rather do that as a followup PR

Reply:
The required change is changing "threaded" to "simple" where the action is called and removing the locking code and the extra scope.  (And changing the function arguments.  Probably just removing the lock, but I'm not sure.)